CMake/External_OpenCV.cmake

```diff
@@ -173,16 +208,17 @@ else()
   list(APPEND OpenCV_EXTRA_BUILD_FLAGS -DBUILD_JPEG=ON)
 endif()
 
-if (OpenCV_SELECT_VERSION VERSION_EQUAL 2.4.11)
+
+# If a patch file exists for this version, apply it
+set (OpenCV_patch ${fletch_SOURCE_DIR}/Patches/OpenCV/${OpenCV_SELECT_VERSION})
+if (EXISTS ${OpenCV_patch})
   set(OPENCV_PATCH_COMMAND ${CMAKE_COMMAND}
-    -DOpenCV_patch:PATH=${fletch_SOURCE_DIR}/Patches/OpenCV2
+    -DOpenCV_patch:PATH=${OpenCV_patch}
     -DOpenCV_source:PATH=${fletch_BUILD_PREFIX}/src/OpenCV
-    -P ${fletch_SOURCE_DIR}/Patches/OpenCV2/Patch.cmake)
+    -P ${OpenCV_patch}/Patch.cmake
+    )
 else()
-  set(OPENCV_PATCH_COMMAND ${CMAKE_COMMAND}
-    -DOpenCV_patch:PATH=${fletch_SOURCE_DIR}/Patches/OpenCV3
-    -DOpenCV_source:PATH=${fletch_BUILD_PREFIX}/src/OpenCV
-    -P ${fletch_SOURCE_DIR}/Patches/OpenCV3/Patch.cmake)
+  set(OPENCV_PATCH_COMMAND "")
```

Comment:
Edit: Having said the above, is this else here in case the user switches OpenCV versions?